opal/mca/smsc/cma/smsc_cma_module.c

```diff
@@ -58,6 +66,26 @@ mca_smsc_endpoint_t *mca_smsc_cma_get_endpoint(opal_proc_t *peer_proc)
         return NULL;
     }
 
+#if OPAL_CMA_KCMP_AVAIL
+    /* Check if CAP_SYS_PTRACE capability is allowed between these two processes
+     * Calling process_vm_readv/writev requires CAP_SYS_PTRACE. We can use kcmp
+     * to check if these two processes share a kernel resource. Since kcmp
+     * also requires CAP_SYS_PTRACE it is a good proxy for process_vm_readv/writev.
+     */
+    rc = syscall(SYS_kcmp, getpid(), modex->pid, KCMP_IO, 0, 0);
```

Comment:
Just curious: Can there be a case where `CAP_SYS_PTRACE` is available and we can use `process_vm_readv` but shared IO context is not? `KCMP_IO` only checks for shared IO contexts:

```
KCMP_IO
              Check whether the processes share I/O context.  The
              arguments idx1 and idx2 are ignored.  See the discussion
              of the CLONE_IO flag in [clone(2)](https://man7.org/linux/man-pages/man2/clone.2.html).
```

Should the check below test explicitly for `-1` (the error case) instead? That should occur when `CAP_SYS_PTRACE` is not available.

Reply:
Good point. I wonder if `KCMP_VM` is more of what we want.

The return value check should be `rc < 0` or `-1 == rc` since it can return a set of positive values (0, 1, 2, 3) in addition to `-1` on error.

Let me run a few experiments then I'll update the code.